lib/cli/src/commands/create_exe.rs

```diff
@@ -344,1043 +256,1624 @@ impl CreateExe {
+            })
+            .canonicalize()
+            .expect("could not find wasmer_main.c / wasmer_main.o")
+            .display()
+    )));
+
+    if zig_triple.contains("windows") {
+        let mut winsdk_path = library_path.clone();
+        winsdk_path.pop();
+        winsdk_path.pop();
+        winsdk_path.push("winsdk");
+
+        let files_winsdk = std::fs::read_dir(winsdk_path)
+            .ok()
+            .map(|res| {
+                res.filter_map(|r| Some(normalize_path(&format!("{}", r.ok()?.path().display()))))
+                    .collect::<Vec<_>>()
+            })
+            .unwrap_or_default();
 
-        for atom_name in atom_names.iter() {
-            let atom_name = Self::normalize_atom_name(atom_name);
+        cmd.args(files_winsdk);
+    }
 
-            write!(
-                c_code_to_instantiate,
-                "
+    println!("{cmd:?}");
 
-            wasm_module_t *atom_{atom_name} = wasmer_object_module_new(store, \"{atom_name}\");
+    let compilation = cmd
+        .stdout(std::process::Stdio::inherit())
+        .stderr(std::process::Stdio::inherit())
+        .output()
+        .context(anyhow!("Could not execute `zig`: {cmd:?}"))?;
 
-            if (!atom_{atom_name}) {{
-                fprintf(stderr, \"Failed to create module from atom \\\"{atom_name}\\\"\\n\");
-                print_wasmer_error();
-                return -1;
-            }}
-            "
-            )
-            .unwrap();
-            write!(deallocate_module, "wasm_module_delete(atom_{atom_name});").unwrap();
-        }
+    if !compilation.status.success() {
+        return Err(anyhow::anyhow!(String::from_utf8_lossy(
+            &compilation.stderr
+        )
+        .to_string()));
+    }
 
-        write!(
-            c_code_to_instantiate,
-            "wasm_module_t *module = atom_{atom_to_run};"
+    // remove file if it exists - if not done, can lead to errors on copy
+    let _ = std::fs::remove_file(&normalize_path(&format!("{}", output_path.display())));
+    std::fs::copy(
+        &normalize_path(&format!("{}", out_path.display())),
+        &normalize_path(&format!("{}", output_path.display())),
+    )
+    .map_err(|e| {
+        anyhow::anyhow!(
+            "could not copy from {} to {}: {e}",
+            normalize_path(&format!("{}", out_path.display())),
+            normalize_path(&format!("{}", output_path.display()))
         )
-        .unwrap();
+    })?;
+
+    Ok(())
+}
 
-        WASMER_STATIC_MAIN_C_SOURCE
-            .replace("#define WASI", "#define WASI\r\n#define WASI_PIRITA")
-            .replace("// INSTANTIATE_MODULES", &c_code_to_instantiate)
-            .replace("##atom-name##", &atom_to_run)
-            .replace("wasm_module_delete(module);", &deallocate_module)
+pub(crate) fn normalize_path(s: &str) -> String {
+    if s.starts_with("\\\\?\\") {
+        s.replacen("\\\\?\\", "", 1)
+    } else {
+        s.to_string()
     }
```

Comment:
It might be easier to write this with `strip_prefix()`.

```suggestion
    s.strip_prefix(r"\\?\").unwrap_or(s).to_string()
```

Reply:
Implemented in https://github.com/wasmerio/wasmer/pull/3430/commits/de39d61937932fe4081fc2931e6f7dcdfd1fe291